### PR TITLE
Restore and wrap up distributed streams module

### PR DESCRIPTION
**Pull request by pwliwanow**

### CODE REVIEW

File: chopsticks-dstream/src/main/scala/dev/chopsticks/dstream/DstreamState.scala [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        _ <- UIO(mapGauge.set(map.size.toDouble))
+      } yield ()
+      _ <- updateQueueGauge.repeat(Schedule.fixed(500.millis.toJava)).fork.toManaged_
+    } yield new Service[Req, Res] {
+      import akkaService.{actorSystem, dispatcher}
+
+      def enqueueWorker(in: Source[Res, NotUsed], metadata: Metadata): UIO[Source[Req, NotUsed]] = {
+        val (inFuture, inSource) = in.watchTermination() { case (_, f) => f }.preMaterialize()
+
+        val completionSignal = Task.fromFuture(_ => inFuture).fold(_ => (), _ => ())
```
Comment (nktpro) [inline chopsticks-dstream/src/main/scala/dev/chopsticks/dstream/DstreamState.scala:58]: A bit cleaner to make it `Task.fromFuture(_ => inFuture).ignore`

File: chopsticks-dstream/src/main/scala/dev/chopsticks/dstream/Dstreams.scala [hunk 2]
```diff
@@ -54,16 +54,19 @@ object Dstreams extends LoggingContext {
...
-        .log("dstream server teardown")
-        .map(_ => ())
-        .catchAll(e => ZLogger.error("Failed unbinding dstream server", e))
-    }
+    ZManaged
+      .make(
+        acquire
+          .logResult("dstream server startup", b => s"Dstream server bound: ${b.localAddress}")
+      ) { binding =>
+        ZIO.access[IzLogging](_.get).map(_.zioLogger).flatMap { zLogger =>
```
Comment (nktpro) [inline chopsticks-dstream/src/main/scala/dev/chopsticks/dstream/Dstreams.scala:62]: `ZIO.access[IzLogging](_.get.zioLogger)` should work

File: chopsticks-sample/src/main/scala/dev/chopsticks/sample/app/GrpcAkkaMasterApp.scala [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+import dev.chopsticks.metric.{MetricCounter, MetricGauge}
+import dev.chopsticks.sample.app.proto.grpc_akka_master_worker._
+import dev.chopsticks.stream.ZAkkaStreams
+import io.prometheus.client.{Counter, Gauge}
+import zio._
+
+import scala.concurrent.duration._
+import scala.jdk.DurationConverters.ScalaDurationOps
+
+object GrpcAkkaMasterApp extends AkkaDiApp[Unit] {
```
Comment (nktpro) [inline chopsticks-sample/src/main/scala/dev/chopsticks/sample/app/GrpcAkkaMasterApp.scala:28]: Let's name this `DstreamsSampleMasterApp` instead, probably also put it in its own `dev.chopsticks.sample.app.dstreams` package together with the worker one

Comment (nktpro) [reply chopsticks-sample/src/main/scala/dev/chopsticks/sample/app/GrpcAkkaMasterApp.scala:28]: Also we can move the other `DstreamSampleApp` in there too. Later we'll organize the other FDB sample apps the same way.

File: chopsticks-dstream/src/main/scala/dev/chopsticks/dstream/DstreamStateMetrics.scala [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package dev.chopsticks.dstream
+
+import dev.chopsticks.metric.{MetricCounter, MetricGauge}
+
+trait DstreamStateMetrics {
```
Comment (nktpro) [inline chopsticks-dstream/src/main/scala/dev/chopsticks/dstream/DstreamStateMetrics.scala:5]: Let's also have Factory and Manager for this Metrics trait as usual

File: chopsticks-dstream/src/main/scala/dev/chopsticks/dstream/DstreamState.scala (hunk 1 above)
Comment (pwliwanow) [reply chopsticks-dstream/src/main/scala/dev/chopsticks/dstream/DstreamState.scala]: Done

File: chopsticks-dstream/src/main/scala/dev/chopsticks/dstream/Dstreams.scala (hunk 2 above)
Comment (pwliwanow) [reply chopsticks-dstream/src/main/scala/dev/chopsticks/dstream/Dstreams.scala]: Done

File: chopsticks-dstream/src/main/scala/dev/chopsticks/dstream/DstreamStateMetrics.scala (hunk 4 above)
Comment (pwliwanow) [reply chopsticks-dstream/src/main/scala/dev/chopsticks/dstream/DstreamStateMetrics.scala:15]: Done

File: chopsticks-sample/src/main/scala/dev/chopsticks/sample/app/GrpcAkkaMasterApp.scala (hunk 3 above)
Comment (pwliwanow) [reply chopsticks-sample/src/main/scala/dev/chopsticks/sample/app/GrpcAkkaMasterApp.scala]: Done

File: chopsticks-testkit/src/main/scala/dev/chopsticks/testkit/ManagedProps.scala [hunk 5]
```diff
@@ -0,0 +1,21 @@
+package dev.chopsticks.testkit
+
+import com.typesafe.config.ConfigFactory
+import zio.{UIO, ZIO, ZManaged}
+
+trait ManagedProps {
```
Comment (nktpro) [inline chopsticks-testkit/src/main/scala/dev/chopsticks/testkit/ManagedProps.scala:6]: Probably rename this to `ManagedSystemProperties` for better clarify

File: chopsticks-sample/src/test/scala/dev/chopsticks/sample/dstreams/DstreamsMasterWorkerTest.scala [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+
+            result <- forkedResult.join
+          } yield assert(result)(equalTo(masterConfig.expected))
+        }
+        managedZio.use(identity)
+      } @@ timeout(20.seconds.toJava)
+    ).provideSomeLayer[LiveRunnableEnv].apply(extraLayer.mapError(TestFailure.fail))
+  }
+
+  private def extraLayer = {
```
Comment (nktpro) [inline chopsticks-sample/src/test/scala/dev/chopsticks/sample/dstreams/DstreamsMasterWorkerTest.scala:115]: Can we leverage / unify `AkkaDiApp` here for building layers as well as bootstrapping actor system instead?

File: chopsticks-sample/src/main/resources/dev/chopsticks/sample/app/dstreams/dstreams-sample-worker-app.conf [hunk 7]
```diff
@@ -0,0 +1,23 @@
+akka {
+  coordinated-shutdown.run-by-jvm-shutdown-hook = on
```
Comment (nktpro) [inline chopsticks-sample/src/main/resources/dev/chopsticks/sample/app/dstreams/dstreams-sample-worker-app.conf:2]: `include required(classpath("app-shared.conf"))` instead, which already sets this?

File: chopsticks-sample/src/main/resources/dev/chopsticks/sample/app/dstreams/dstreams-sample-worker-app.conf [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+  http.server.preview.enable-http2 = on
+
+  log-dead-letters = off
+  loglevel = INFO
+
+  stream.materializer.debug.fuzzing-mode = off
```
Comment (nktpro) [inline chopsticks-sample/src/main/resources/dev/chopsticks/sample/app/dstreams/dstreams-sample-worker-app.conf:9]: This should already be `off` by default and it's only set to `on` under test `resources`

File: chopsticks-sample/src/main/resources/dev/chopsticks/sample/app/dstreams/dstreams-sample-worker-app.conf [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+}
+
+iz-logging {
```
Comment (nktpro) [inline chopsticks-sample/src/main/resources/dev/chopsticks/sample/app/dstreams/dstreams-sample-worker-app.conf:12]: Should already inherit from https://github.com/shopstic/chopsticks/blob/142eff5c6855e447749df7ed9884446b4b78b076/chopsticks-fp/src/main/resources/reference.conf#L13

File: chopsticks-sample/src/test/scala/dev/chopsticks/sample/dstreams/DstreamsMasterWorkerTest.scala (hunk 6 above)
Comment (pwliwanow) [reply chopsticks-sample/src/test/scala/dev/chopsticks/sample/dstreams/DstreamsMasterWorkerTest.scala]: There is now `AkkaDiRunnableSpec` that reuses whatever we can from `AkkaDiApp` and can be used for Akka and DiStage based integration tests

Comment (pwliwanow) [reply chopsticks-sample/src/test/scala/dev/chopsticks/sample/dstreams/DstreamsMasterWorkerTest.scala]: @nktpro please let me know if `AkkaDiRunnableSpec` should take type parameter (that would be a config) and whether `diTestM` should accept additional customize method: `customizeConfig: TestConfig => TestConfig`

**Pull request merged**